packages/block-editor/src/hooks/layout.js

```diff
@@ -265,6 +268,66 @@ function LayoutPanel( { setAttributes, attributes, name: blockName } ) {
 	);
 }
 
+function ChildLayoutPanel( { setAttributes, attributes } ) {
+	const { style = {} } = attributes;
+	const { layout = {} } = style;
+	const { selfStretch, flexSize } = layout;
+
+	return (
+		<InspectorControls>
+			<PanelBody title={ __( 'Child Layout' ) }>
+				<ToggleGroupControl
+					label={ __( 'Relative size' ) }
+					value={ selfStretch || 'hug' }
+					onChange={ ( value ) => {
+						setAttributes( {
+							style: {
+								layout: {
+									...layout,
+									selfStretch: value,
+								},
+							},
+						} );
+					} }
+					isBlock={ true }
+				>
+					<ToggleGroupControlOption
+						key={ 'hug' }
+						value={ 'hug' }
+						label={ __( 'Hug' ) }
+					/>
+					<ToggleGroupControlOption
+						key={ 'fill' }
+						value={ 'fill' }
+						label={ __( 'Fill' ) }
+					/>
+					<ToggleGroupControlOption
+						key={ 'fixed' }
+						value={ 'fixed' }
+						label={ __( 'Fixed' ) }
+					/>
+				</ToggleGroupControl>
+				{ selfStretch === 'fixed' && (
+					<UnitControl
+						onChange={ ( value ) => {
+							setAttributes( {
+								style: {
+									...style,
+									layout: {
+										...layout,
+										flexSize: value,
```

Comment:
Is `flexSize` always going to be output as `flex-basis`? If so, I was wondering if we can use the latter, since the `style` object tries to stick closer to CSS properties where possible (where the root `layout` object appears to currently be a bit more abstract).

Reply:
I was trying to avoid giving it the property name in favour of a name that's clearer as to what it does, mainly because `flexSize` translates into more than one property (right now it's `flex-shrink` and `flex-basis`, but we might add `border-box` as per your suggestion, things might change etc.

Reply:
> I was trying to avoid giving it the property name in favour of a name that's clearer as to what it does, mainly because flexSize translates into more than one property

Good point! Yes, for these ones that result in multiple / combined rules being output (or additional logic) that makes good sense to me, a bit like how `blockGap` doesn't directly correspond to `gap` 👍

I've also been wondering when it makes most sense to store things in the layout object directly or when to store them in `style.layout`. I quite like the decision to use `style.layout` here, personally, but I wondered if you had an idea in mind of when we make the call between the two? One possibility is when we aren't selecting from a controlled list of options (so things that aren't alignment based like content justification), `style.layout` might be the better fit?

Reply:
My idea here was to keep the `layout` attribute exclusively for blocks that have layout themselves, and use `style.layout` for styles that are related to a parent's layout.